src/arm7tdmi.rs

```diff
@@ -93,82 +95,113 @@ impl Arm7tdmi {
         todo!("Branch Link")
     }
 
-    fn mov(&mut self, op_code: u32) {
-        // bits [24-21] are the RD
-        let rd = (op_code & 0x00_00_F0_00) >> 12;
-        println!("RD: {:?}", rd);
-
-        // 25th bit is I = Immediate Flag
-        let immediate: bool = (op_code & 0x02_00_00_00) >> 25 == 1;
-        println!("Immediate: {:?}", immediate);
-
-        // 20th bit is S = Condition Set
-        if op_code & 0x00_08_00_00 > 0 {
-            todo!("Condition set")
-        }
-
-        if immediate {
-            // bits [7-0] are the immediate value
-            let immediate_value = op_code & 0x00_00_00_FF;
-            println!("value: {:?}", immediate_value);
+    fn data_processing(&mut self, op_code: u32) {
+        /// bit [25] is I = Immediate Flag
+        let i = ((op_code & 0x02_00_00_00) >> 25) as u8;
+        /// bit [24-21]
+        let alu_opcode = ((op_code & 0x01_E0_00_00) >> 25) as u8;
+        /// bit [20] is sets condition codes
+        let s = ((op_code & 0x00_10_00_00) >> 20) as u8;
+        /// bits [15-12] are the Rd
+        let rd = ((op_code & 0x00_00_F0_00) >> 12) as u8;
+
+        let op2 = match i {
+            /// Register as 2nd Operand
+            0 => {
+                // Shift Type (0=LSL, 1=LSR, 2=ASR, 3=ROR)
+                let shift_type = ((op_code & 0x00_00_00_60) >> 5) as u8;
+                // bit [4] is Shift by Register Flag (0=Immediate, 1=Register)
+                let r = (op_code & 0x00_00_00_10) >> 4;
+                // 2nd Operand Register (R0..R15) (including PC=R15)
+                let mut op2 = ((op_code & 0x00_00_00_0F) >> 8);
+
+                match r {
+                    /// Shift by amount
+                    0 => {
+                        /// Shift amount
+                        let is = ((op_code & 0x00_00_07_80) >> 7) as u8;
+                        match is {
+                            0 => match shift_type {
```

Comment:
I'm asking if is better to split the nested match for readability at least but also for testing

Reply:
I am aware of this, however, it is still not the full implementation, there is room for improvement